README.md

```diff
@@ -97,6 +97,69 @@ Book::STATES['in_weird_state'] # 'in weird State'
 
 ```
 
+#### NxtSupport::HashTranslator
+
+`NxtSupport::HashTranslator` is a module that allows you to manipulate keys and values of original hash through tuple hash. 
+Tuple hash is a hash where `key` - represent's the key of original hash, and `value` - represents the key of result hash.
+Use `#translate_hash` method to get the result hash.
+
+```ruby
+class TestClass
+  include NxtSupport::HashTranslator
+end
+
+TestClass.translate_hash(firstname: 'John', firstname: :first_name)
+=> { 'first_name' => 'John' }
+```
+The `value` also could be a `Hash`. If this hash contain only 1 pair `key` would be used as a new `key`
+and `value` would be used as it is unless it responds to `#call`. In this cases `#call` would be called and actual value
+passed to the block called (if no, original value would be left). In case there is more than 1 pair first `key` would
+be used and last `value` would be used as well. 
+
+```ruby
+class TestClass
+  include NxtSupport::HashTranslator
+end
+# Multiple pair hash used as hash value
+hash = { firstname: 'John', phonenumber: '11-22-33-445' }
+tuple = {
+          firstname: :first_name,
+          phonenumber: {
+            phone_number: ->(number) { number.to_s.prepent('+49') },
+            telephone: :phone_num
```

Comment:
But that is indeed weird that there is also `telephone: :phone_num` even though there is no :telephone in the input? Think we should make simple understandable examples and probably explain the different cases.

Reply:
I just want to show the behaviour. In case you pass (for any reason) the hash with more than key-value pair it would act so. Why it was build in such manner yet unknown, but if I change it needs to be checked all across the services.
BTW, it looks like you've implemented this [logic](https://github.com/nxt-insurance/application-service/pull/64/files#diff-01d3410d61e41100973c525fece922d4), did you?